[v4] Refactor Proxy documentation (#475)
strapi/documentation: docs/developer-docs/latest/setup-deployment-guides/deployment/optional-software/haproxy-proxy.md

The below examples are more or less acting as an "SSL termination" proxy, meaning that HAProxy is only accepting the requests on SSL and proxying to other backend services such as Strapi or other web servers. **HAProxy cannot serve static content** and as such it is usually used to handle multi-server deployments in a failover or load-balance situation. The examples provided below are based around everything existing on the same server, but could easily be tweaked for multi-server deployments.

!!!include(developer-docs/latest/setup-deployment-guides/deployment/optional-software/snippets/strapi-server.md)!!!

### HAProxy

As mentioned previously the following examples are either proxying all requests directly to Strapi or are splitting requests between Strapi and some other backend web server such as Nginx, Apache, or others.

Below are 3 example HAProxy configurations:

- Sub-domain based such as `api.example.com`
- Sub-folder based with both the API and Admin on the same sub-folder such as `example.com/test/api` and `example.com/test/admin`
- Sub-folder based with split API and Admin such as `example.com/api` and `example.com/dashboard`

> :::warning HAProxy SSL Support

Reviewer: ```suggestion
:::caution HAProxy SSL Support
```

---

> - Example Domain: `api.example.com`

Reviewer: ```suggestion
- Example domain: `api.example.com`
```

> - Example Admin: `api.example.com/admin`

Reviewer: ```suggestion
- Example admin: `api.example.com/admin`
```

- Example API: `api.example.com/api`

> - Example Uploaded Files (local provider): `api.example.com/uploads`

Reviewer: ```suggestion
- Example uploaded Files (local provider): `api.example.com/uploads`
```

#### Sub-Folder Unified

> This config is using a sub-folder that is dedicated to Strapi only. It will redirect normal HTTP traffic over to SSL and proxies the "frontend" to `localhost:8080`, but proxies all Strapi requests on the `example.com/test` sub-path to the locally running Strapi application.

Reviewer: ```suggestion
This configuration is using a sub-folder dedicated to Strapi only. It will redirect normal HTTP traffic over to SSL and proxies the front end to `localhost:8080`, but proxies all Strapi requests on the `example.com/test` sub-path to the locally running Strapi application.
```

:::caution

> HAProxy **cannot** serve static content, the below example is proxying frontend traffic to some other web server running on the localhost port 8080

Reviewer: ```suggestion
HAProxy cannot serve static content, the below example is proxying front-end traffic to some other web server running on the localhost port 8080.
```

:::

---

> - Example Domain: `example.com/test`

Reviewer: ```suggestion
- Example domain: `example.com/test`
```

> - Example Admin: `example.com/test/admin`

Reviewer: ```suggestion
- Example admin panel: `example.com/test/admin`
```

- Example API: `example.com/test/api`

> - Example Uploaded Files (local provider): `example.com/test/uploads`

Reviewer: ```suggestion
- Example uploaded Files (local provider): `example.com/test/uploads`
```

---

> - Example Domain: `example.com`

Reviewer: ```suggestion
- Example domain: `example.com`
```

> - Example Admin: `example.com/dashboard`

Reviewer: ```suggestion
- Example admin panel: `example.com/dashboard`
```

- Example API: `example.com/api`

> - Example Uploaded Files (local provider): `example.com/uploads`

Reviewer: ```suggestion
- Example uploaded Files (local provider): `example.com/uploads`
```
